src/runtime/recipe/handle-connection.ts

```diff
@@ -78,31 +78,34 @@ export class HandleConnection {
     return 0;
   }
 
-  get recipe() { return this._recipe; }
-  get name() { return this._name; } // Parameter name?
-  getQualifiedName() { return `${this.particle.name}::${this.name}`; }
-  get tags() { return this._tags; }
+  get recipe(): Recipe { return this._recipe; }
+  get name(): string { return this._name; } // Parameter name?
+  getQualifiedName(): string { return `${this.particle.name}::${this.name}`; }
+  get tags(): string[] { return this._tags; }
+
   get type() {
     if (this.resolvedType) {
       return this.resolvedType;
     }
     const spec = this.spec;
     return spec ? spec.type : null;
   }
-  get direction() { // in/out/inout/host/consume/provide
+
+  get direction(): Direction|undefined { // in/out/inout/host/consume/provide
     if (this._direction) {
       return this._direction;
     }
     const spec = this.spec;
     return spec ? spec.direction : null;
   }
-  get isInput() {
+
+  get isInput(): boolean {
     return this.direction === 'in' || this.direction === 'inout';
   }
-  get isOutput() {
+  get isOutput(): boolean {
     return this.direction === 'out' || this.direction === 'inout';
   }
-  get handle() { return this._handle; } // Handle?
+  get handle(): Handle|undefined { return this._handle; } // Handle?
   get particle() { return this._particle; } // never null
 
   set tags(tags: string[]) { this._tags = tags; }
```

Comment:
Should we mark setters with `void` returns, or is that too verbose?

Reply:
I think typescript doesn't allow setter returns, they are implicit void.